home/views.py

```diff
@@ -58,20 +59,30 @@ def dashboard(request):
     query_date_str = request.GET.get('d', '')
     query_section = request.GET.get('s', '')
 
+    day_dict = {1: 'Monday', 2: 'Tuesday', 3: 'Wednesday',
+                4: 'Thursday', 5: 'Friday', 6: 'Saturday', 7: 'Sunday'}
+
     if query_date_str and query_section:
         query_date = datetime.strptime(query_date_str, '%Y-%m-%d').date()
         query_day = query_date.strftime("%w")
+        print(query_section)
         calendar = Calendar.objects.filter(date=query_date)
+        subject_today = Schedule.objects.filter(
+            day=query_day, section__section_id=query_section).get()
+        sub = subject_today.sub_dict[subject_today.subject]
```

Comment:
Use proper variable name - `subject_scheduled`. And for getting the subject name, you can use `get_subject_display()` instead.

Reply:
Also, both these above variables should be placed after line 102 (`context['cw_hw'] = cw_hw`) once it's verified that the schedule for that day exists and class is scheduled for that day. Also, after that line, there won't be a need of creating the first variable (`subject_today`) as you can use the already created `schedule` variable.